src/actions/chatXActions.js

```diff
@@ -135,14 +126,15 @@ export function getMessageListStartAtKey(key) {
 
 export function getNewMessages(callback) {
     return (dispatch, getState) => {
-        let msg = {};
-        return firebaseApi.getRef('/chatX/')
-            .on('child_added', (snap) => {
-                if (getState().chatX.isDataLoaded) {
-                    msg = extractMessageProperties(snap.val(), getState().user.email);
-                    callback(msg);
+        return firebaseApi.getChildAdded('/chatX/', (snap) => {
+            if (getState().chatX.isDataLoaded) {
+                if (getState().chatX.firstKey) {
+                    dispatch(getMessageListStartAtKey(getState().chatX.firstKey));
+                } else {
+                    dispatch(getInitial10Messages());
```

Comment:
handle case when there was no message when entering the room and hence no firstKey